keps/sig-instrumentation/distributed-tracing-kep.md

```diff
@@ -0,0 +1,134 @@
+
+In addition, due to the self-healing nature of Kubernetes, regressions wherein latencies are affected but the overall task is eventually accomplished are not uncommon. With our current monitoring architecture, these "soft regressions" are often difficult to observe and diagnose. Collecting structured trace data on per-object latencies would enable us to detect these long-term regressions automatically, and quickly determine their root causes.
+
+
+### Goals
+
+* Make it possible to visualize the progress of objects across distinct Kubernetes components
+* Streamline the tedious processes involved in debugging latency issues in Kubernetes
+* Make it possible to identify high-level latency regressions, and attribute them to offending processes
+
+
+### Non-Goals
+
+* To replace existing logging, metrics, or the events API
+* To trace operations from all Kubernetes resource types in a generic manner (i.e. without manual instrumentation)
+
+## Proposal
+
+The changes proposed in this KEP consist both of out-of-tree additions and in-tree changes, as described below. 
+
+### In-tree changes
+
+#### Vendoring the OpenCensus trace framework
+
+This KEP proposes the use of the [OpenCensus tracing framework](https://opencensus.io/) to create and export spans to configured backends. The OpenCensus framework was chosen for various reasons: 
+
+1) Provides concrete, tested implementations for creating and exporting spans to diverse backends, rather than providing an API specification, as is the case with [OpenTracing](https://opentracing.io/specification/)
+2) [Provides an agent](https://github.com/census-instrumentation/opencensus-service) which enables lazy configuration for exporters, batching of spans, and other features 
+3) Offers support for context-aware metrics in addition to tracing, which could be used in the future
+
+#### Adding trace utility package
+
+This package should make it straightforward to export spans from Kubernetes components wherein we have access an object's `trace.kubernetes.io/context` annotation. This package should make it trivial to add traces to an object without trace instrumentation, as well as to extend existing trace instrumentation for an object.
+
+[Located here is an example of what such a package could look like.](https://github.com/Monkeyanator/kubernetes/blob/c28bbce4262b8da65e4667129ff7c4b5844a3ab5/pkg/util/trace/traceutil.go) For the final implementation, we must divorce the package from specific exporters, and allow users to configure the desired tracing backend elsewhere.
+
+#### Instrumenting various Kubernetes components with pod startup traces
+
+As we move forward with this KEP, we hope to leverage the aforementioned trace utility package to trace basic pod-related operations across the scheduler and kubelet. A tentative implementation of this instrumentation is [located here](https://github.com/Monkeyanator/kubernetes/pull/15).
```

Comment:
Since this KEP will stay here, you should avoid including things that will change.  I.E. _now_ the KEP isn't merged and work isn't done, but that won't be the case for long.  You should write _as if_ the KEP has been merged, and the implementation has been done.  Just describe what things will look like.